core/src/main/codegen/templates/Parser.jj

```diff
@@ -5554,9 +5558,33 @@ SqlNode BuiltinFunctionCall() :
         }
         (
             <USING> name = SimpleIdentifier() {
+<#if parser.allowTranslateUsingCharSetWithError>
+                String nameString = name.toString().toUpperCase();
+                String[] charSets = new String[2];
+                if(nameString.contains("_TO_") && nameString.split("_TO_").length == 2){
+                    allowTranslateUsingCharSet = true;
+                    charSets = nameString.split("_TO_");
+                    args.add(new SqlCharacterSetToCharacterSet(charSets, getPos()));
+                }else{
+                    args.add(name);
+                }
+<#else>
                 args.add(name);
+</#if>
             }
+<#if parser.allowTranslateUsingCharSetWithError>
+            [
+              <WITH> <ERROR> {
+                isWithError = true;
+              }
+            ]
+</#if>
             <RPAREN> {
+<#if parser.allowTranslateUsingCharSetWithError>
+                if(allowTranslateUsingCharSet){
```

Comment:
nit: missing space before open paren and after close paren

Reply:
Fix in https://github.com/googleinterns/calcite/pull/84/commits/8f1e577ebd71a0824d776563241bad47007558e9